sdk/storage/azure-storage-blob/tests/test_blob_client.py

```diff
@@ -438,6 +439,31 @@ def test_create_service_with_custom_account_endpoint_path(self):
         self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
         self.assertEqual(service.url, 'http://local-machine:11002/custom/account/path/foo/bar?snapshot=baz')
 
+    def test_create_blob_client_with_sub_directory_path_in_blob_name(self):
+        blob_url = "https://testaccount.blob.core.windows.net/containername/dir1/sub000/2010_Unit150_Ivan097_img0003.jpg"
+        blob_client = BlobClient.from_blob_url(blob_url)
+        self.assertEqual(blob_client.container_name, "containername")
+        self.assertEqual(blob_client.blob_name, "dir1/sub000/2010_Unit150_Ivan097_img0003.jpg")
+
+        blob_emulator_url = 'http://127.0.0.1:1000/devstoreaccount1/containername/dir1/sub000/2010_Unit150_Ivan097_img0003.jpg'
+        blob_client = BlobClient.from_blob_url(blob_emulator_url)
+        self.assertEqual(blob_client.container_name, "containername")
+        self.assertEqual(blob_client.blob_name, "dir1/sub000/2010_Unit150_Ivan097_img0003.jpg")
+
+    def test_create_client_for_emulator(self):
+        container_client = ContainerClient(
+            account_url='http://127.0.0.1:1000/devstoreaccount1',
+            container_name='newcontainer',
+            credential='Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==')
```

Comment:
hoping this is not real :P

Reply:
it's not

Reply:
I bet we'll get email to remove it soon :P